### PR TITLE
allow selectively enabling/disabling controllers

### PR DESCRIPTION
**Pull request by joelddiaz**
Introduce `--controllers` to allow specifying a comma-separated list of controllers to run.
Introduce `--disabled-controllers` to allow specifying a comma-separated list of controllers to not run (disabled takes precedence over anything in `--controllers`).

Modify how controllers are registered so that we can use the information from these new parameters when initializing controllers on startup.

This should allow a development workflow where you disable the controller(s) you are working on in your "deployed" hive instance, and then only enable the controller(s) you are working on for you local run/debug session. (You do need to remember to disable leader election for your local run.)

Note: the current controller names have been left unchanged, and there may be a desire to standardize on a naming scheme.

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -150,12 +202,28 @@ func newRootCommand() *cobra.Command {
...
 }
 
+func newControllerManagerOptions() *controllerManagerOptions {
+	// By default we have all of the controllers enabled
+	controllers := make([]string, len(controllerFuncs))
+	i := 0
+	for name := range controllerFuncs {
+		controllers[i] = name
+		i++
+	}
```
Comment (staebler) [inline cmd/manager/main.go:221]:
```suggestion
	controllers := make([]string, 0, len(controllerFuncs))
	for name := range controllerFuncs {
		controllers = append(controllers, name)
	}
```

**Comment by joelddiaz** [comment]
incorporated @staebler 's recommendation. i can just go ahead and squash if this is ready.

**Comment by staebler** [comment]
> incorporated @staebler 's recommendation. i can just go ahead and squash if this is ready.

Squash away, as far as I am concerned.

**Comment by staebler** [comment]
There are merge conflicts.

/lgtm cancel

**Comment by joelddiaz** [comment]
@staebler yes, i had to update for the removal of the clusterreolcate bits. should be good again.

**Pull request merged**